### DISCUSSION TITLE
A global hook that allows modification of a result sql query

### DISCUSSION THREAD

**Discussion by JasonRammoray** [mikro-orm/mikro-orm#4861] [Ideas]
**Is your feature request related to a problem? Please describe.**
Assume an application stores data in `PostgreSQL` and has multiple `MikroORM` models, some of which have [text](https://www.postgresql.org/docs/current/datatype-character.html) fields.
An attempt to store `\u0000` character as part of that `text` field's value will render an app crashed, because it's unsupported by `PostgreSQL`.

**Describe the solution you'd like**
A global hook, which allows to override a result `sql` query.
Pseudocode:
```typescript
MikroOrm.registerHooks({
   onDbQueryReady(query: string) {
       return query.replace(/[\u0000-\uffff]/g, '');
   },
});
```

**Describe alternatives you've considered**
One of the possible fixes is to employ [BeforeCreate](https://mikro-orm.io/docs/5.6/decorators#beforecreate) and [BeforeUpdate](https://mikro-orm.io/docs/5.6/decorators#beforeupdate) decorators on the respective models.
Although that could work, it's laborious.

**Comment by B4nan** [comment]
Why dont you just use a custom type that maps the value? Sounds weird to have a hook for this, do you know any other tool that would have that?

**Reply by JasonRammoray** [reply]
I guess a custom type could do the job as well, although that involves quite some changes.
That is why I was thinking of something like [before_execute](https://docs.sqlalchemy.org/en/13/core/events.html#sqlalchemy.events.ConnectionEvents.before_execute) in `SQLAlchemy`.

**Reply by B4nan** [reply]
You can alter how the default type mapping works globally:

https://mikro-orm.io/docs/configuration#adjusting-default-type-mapping

**Reply by JasonRammoray** [reply]
Oh, I see.
I'll try to explore that approach with a globally mapped custom type.
I guess we can close the discussion at that point.
Thanks for the hint.
